### PR TITLE
doc: extensions: doxyrunner: add doxyrunner_outdir_var option

### PR DESCRIPTION
**Pull request by gmarull**
The doxyrunner_outdir_var option allows to specify which variable (if
any) is used to represent the output directory (as used by
OUTPUT_DIRECTORY). This makes sure that other options referencing it are
processed correctly, since the output directory is not passed as a
variable.

This new option can be used to e.g. generate a tag file.

### CODE REVIEW

File: doc/_extensions/zephyr/doxyrunner.py
Comment (JordanYates) [inline doc/_extensions/zephyr/doxyrunner.py:138]: Should this be `Optional[Path]`?

**Review by JordanYates — APPROVED** [review]
Otherwise looks good, I can confirm it generates the tag file into the desired directory
```
jordan@FIDO:~/code/zephyr$ ls -lah zephyr/doc/_build/html/doxygen/html/zephyr.tag
-rw-r--r-- 1 jordan jordan 8.8M Jan 11 08:14 zephyr/doc/_build/html/doxygen/html/zephyr.tag
```

**Comment by JordanYates** [comment]
Once approved I was also going to ask if my PR could be backported to the v2.7 branch.
You may as well cherry-pick my last commit onto this PR so it's all grouped together.

File: doc/_extensions/zephyr/doxyrunner.py
Comment (gmarull) [reply doc/_extensions/zephyr/doxyrunner.py:138]: No, since it is a variable name, not its content (`DOXY_OUT` in this case). I know it's not intuitive...

**Comment by gmarull** [comment]
I have dropped the `GENERATE_TAG` addition, since this will be added by https://github.com/zephyrproject-rtos/zephyr/pull/41530

**Review by mbolivar-nordic — APPROVED** [review]
Seems fine

**Comment by gmarull** [comment]
@nashif ping

**Review by nashif — APPROVED** [review]

**Pull request merged**
